### ISSUE TITLE
Exceptions thrown by RenderingAndSnapshot.equals are swallowed.

### ISSUE THREAD

**Issue by rjrjr** [square/workflow-kotlin#224]
The `fail()` call in the test below is swallowed.

To see this exact failure, you'll have to revert the fix for #223 if that has landed. Or else replace the String rendering type with something that will throw from `equals`.

```kotlin
    lateinit var sink: Sink<String>

    val workflow = Workflow.stateful<Unit, String, Nothing, String>(
        initialState = { _, _ -> "unchanging state" },
        snapshot = { Snapshot.of {
          fail("Snapshot must not be resolved.")
        } },
        render = { _, state ->
          sink = actionSink.contraMap { action { this.state = it } }
          state
        }
    )
    val props = MutableStateFlow(Unit)
    val renderings = renderWorkflowIn(workflow, scope, props) {}

    renderings.value.rendering.let { rendered ->
      assertEquals("unchanging state", rendered)
      sink.send("unchanging state")
    }
```

**Comment by zach-klippenstein** [comment]
Yikes. Surprised we don't have tests around this either.

**Comment by zach-klippenstein** [comment]
@rjrjr If I understand correctly, this is fixed by #225 because renderings are now never compared by equals? I'll still add unit tests around rendering and snapshot equals to make sure nothing else is eating them up.
